### DISCUSSION TITLE
Unsuccessful connecting to VerneMQ over secure websockets (wss) -- what options need to be set?

### DISCUSSION THREAD

**Discussion by matthewadams** [mqttjs/MQTT.js#1832] [Q&A] [answered]
We're using mqttjs 4.3.7 with Node.js 18.17.1 from the browser (quasar+vite+vuejs single-page application), and we're having trouble getting the `MqttClient` to successfully connect.  We've tried many different combinations of options, and none of them work.  We just get a timeout.  Other clients successfully connect:  the `mqttx` cli client, the MQTTX gui, even https://mqttjs.github.io/mqtt-elements/ connects successfully.

Our wss url is similar to `wss://the.server.name/mqtt`, and the connection options we're passing are (some values redacted):
```
{
    "clean": true,
    "keepalive": 30,
    "clientId": "[admin-ui][px4rghpqnuepi7s34hs9wu458jj]",
    "connectTimeout": 15000,
    "reconnectPeriod": 1000,
    "username": "(redacted)",
    "password": "(redacted)",
    "protocolVersion": 5,
    "protocolId": "MQTT"
  }
```
Our server certificate is signed by a well-known authority, and the browsers are happy with it; they happily serve the rest of the site with no security warnings.  Here's a connction code snippet:
```
    const client: mqtt.MqttClient = await new Promise((resolve, reject) => {
      const c = mqtt.connect(wsUrl, mqttOpts)
      c.on("connect", () => {
        millis = Date.now() - then
        c.removeAllListeners()
        resolve(c)
      })
      c.on("error", (e) => {
        millis = Date.now() - then
        c.removeAllListeners()
        reject(e)
      })
      c.on("offline", () => {
        millis = Date.now() - then
        c.removeAllListeners()
        reject(new Error("disconnected"))
      })
    })
```
After the timeout, there is no more information.

I'm also trying to test with the mqtt cli client, but there doesn't appear to be a way to specify the path on the host or the full url to test publishing a message.  I have [a separate discussion](https://github.com/mqttjs/MQTT.js/discussions/1831) regarding that.

Any help is appreciated.

**Comment by robertsLando** [comment]
> We're using mqttjs 4.3.7 with Node.js 18.17.1 from the browser

Firstly upgrade to latest 5.5.0. Also please open an issue type of bug and fill all the fields requested (including logs)

**Comment by matthewadams** [comment]
Upgrading to 5.5.0 would require us to upgrade our Node.js version due to `bigint` issues.  What's the minimum Node.js version required for 5.5.0?

**Comment by matthewadams** [comment]
I see at https://github.com/mqttjs/MQTT.js/blob/v5.5.0/package.json#L101
```
  "engines": {
    "node": ">=16.0.0"
  },
```
Is that correct?  If so I'm going to have to dig a little into this `bigint` business.

**Reply by robertsLando** [reply]
Yeah that's correct!

**Comment by matthewadams** [comment, marked as answer]
After some internetterwebby searching, we were able to get the error to go away by adding the statement below, where we set `viteConf.optimizeDeps` in `quasar.config.js`:
```
      extendViteConf(viteConf) {
        viteConf.resolve.preserveSymlinks = true
        viteConf.optimizeDeps = { /////// <= added this
          esbuildOptions: {
            supported: {
              bigint: true
            }
          }
        }
      },
```
This allowed us to start using `5.5.0`.  Hope this helps others.
